app/controllers/jobs_controller.rb

```diff
@@ -1,12 +1,50 @@
 class JobsController < ApplicationController
   def index
-    @jobs = jobs_payload
+    @jobs = Job.all
+    render json: {jobs: @jobs}
+  end
+
+  def my_jobs
+    @jobs = current_user.jobs + Job.where(provider_id: current_user.id)
+    render json: {jobs: @jobs}
   end
 
   def new
     render 'new.html.erb'
   end
 
+  def create
+    @job = Job.new(
+      requested_time: params[:requested_time],
+      user_id: current_user.id,
+      status: "posted"
+      )
+    @job.save
+
+    render json: {job: @job}
+  end
+
+  def update
+    @job = Job.find(params[:id])
+
+    @job.update(
+      requested_time: params[:requested_time] || @job.requested_time,
+      start_time: params[:start_time] || @job.start_time,
+      end_time: params[:end_time] || @job.end_time,
+      provider_id: params[:provider_id] || @job.provider_id,
+      status: params[:status] || @job.status
+      )
+
+    render json: {job: @job}
+    
```

Comment:
See comment above for `StrongParameters`. please update the `update` action to use strong params

See comments above for errors